SwiftDeck/SwiftDeck/Communication/CommunicationManager.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  CommunicationManager.swift
+//  SwiftDeck
+//
+//  Created by Ilija Puaca on 22/1/20.
+//  Copyright © 2020 Unfolded. All rights reserved.
+//
+
+import WebKit
+
+/// Error type that's thrown when a message payload is invalid.
+enum InvalidMessageError: LocalizedError {
+
+    case invalidData
+
+    // MARK: - LocalizedError implementation
+
+    var errorDescription: String? {
+        switch self {
+        case .invalidData:
+            return NSLocalizedString("Unable to encode the provided message data", comment: "Invalid message data error message")
+        }
+    }
+
+}
+
+/// Utility object that handles communication with external resources.
+final class CommunicationManager {
```

Comment:
Just noting that you seem to have the beginnings of a generic transport implementation here...

I like the idea of having each of the language bindings have a `transport` directory that just focuses on the message passing and gradually we align these cross-platform.